### PR TITLE
update Mohist egg

### PR DESCRIPTION
**Pull request by cocorico8**
# Description

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
* [x] You verify that the start command applied does not use a shell script
* [x] The egg was exported from the panel

Ok so I know there is already a PR for this one, but mine is refactored for the install script, I based it on the paper install script so it's cleaner, i also added the possibility to install the fabric version of mohist (banner), also added java 21. In this PR I also fixed the proxy readme as it was not pointing to the right place and added the fact that windows 11 is a thing.

### CODE REVIEW

File: java/mohist/egg-mohist.json [hunk 1]
```diff
@@ -15,25 +15,26 @@
...
-        "Java 18": "ghcr.io\/pterodactyl\/yolks:java_18"
+        "Java 11": "ghcr.io\/pterodactyl\/yolks:java_11",
+        "Java 16": "ghcr.io\/pterodactyl\/yolks:java_16",
+        "Java 17": "ghcr.io\/pterodactyl\/yolks:java_17",
+        "Java 18": "ghcr.io\/pterodactyl\/yolks:java_18",
+        "java 21": "ghcr.io\/pterodactyl\/yolks:java_21"
     },
     "file_denylist": [],
-    "startup": "java -Xms128M -XX:MaxRAMPercentage=95.0 -Dterminal.jline=false -Dterminal.ansi=true -jar {{SERVER_JARFILE}}",
+    "startup": "java -Xms128M -Xmx{{SERVER_MEMORY}}M -Dterminal.jline=false -Dterminal.ansi=true -jar {{SERVER_JARFILE}}",
```
Comment (QuintenQVD0) [inline java/mohist/egg-mohist.json:26]: No, this is wrong.

Keep the 95 logic as it send 95% of the allocated ram to the server and works with unlimited (0)

File: java/mohist/egg-mohist.json [hunk 2]
```diff
@@ -44,30 +45,71 @@
...
+            ],
+            "sort": 1,
             "field_type": "text"
         },
         {
             "name": "Minecraft Version",
             "description": "The version of Minecraft to download.",
-            "env_variable": "MC_VERSION",
-            "default_value": "1.19.4",
+            "env_variable": "MINECRAFT_VERSION",
```
Comment (QuintenQVD0) [inline java/mohist/egg-mohist.json:59]: You should never edit the env variable, keep it as is

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: java/mohist/egg-mohist.json (hunk 2 above)
Comment (cocorico8) [reply java/mohist/egg-mohist.json]: updated

File: java/mohist/egg-mohist.json (hunk 1 above)
Comment (cocorico8) [reply java/mohist/egg-mohist.json]: updated

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! thanks

**Pull request merged**